### ISSUE TITLE
Release is desperately needed

### ISSUE THREAD

**Issue by DanielVartanov** [chrisk/fakeweb#57]
The last version `1.3.0` was released **in year 2010**, 6 years ago!
People needed to pull down changes which were made since that. 

Please please please bump up and release a new version.

**Comment by glebm** [comment]
The maintainer of this gem hasn't been active on Github since 2014 and hasn't tweeted nor instagram'd, nor flickr'd anything since November 2015. His homepage is gone as well.

**Comment by DanielVartanov** [comment]
@glebm oh god, are you assuming the worst? Let's hope it's not the case. But someone needs to take over maintenance anyway. I really don't want to undertake for this task as I'll won't devote enough time for the project, but what would you suggest for such case? Is there a common protocol in the community?

**Comment by glebm** [comment]
> oh god, are you assuming the worst?

Not necessarily, as he also seems to have a history of quitting the internet: https://twitter.com/robolson/status/667436321812701189

> Is there a common protocol in the community?

Usually, an org that uses the gem heavily and is willing to maintain it would fork the repo, and email rubygems explaining the situation to get ownership of the gem, and contact github to set the new repo as the canonical one. There are several precedents, such as rake.

**Comment by DanielVartanov** [comment]
@glebm great, huge thanks for explanation

**Comment by betesh** [comment]
> Usually, an org that uses the gem heavily and is willing to maintain it forks the repo, emails rubygems explaining the situation to get ownership of the gem, and contacts github to set the new repo as the canonical one. There are several precedents, such as rake.

Has anyone done that yet?

**Comment by betesh** [comment]
@technicalpickles @MarkMenard @mislav @blaine @jacqui @myronmarston @kachick @illuminerdi @syborg @trevorrowe @larrytheliquid @chrisZingel @josh @Empact @akahn @chubas @anselmoalves 

All of you have contributed to this gem in the past.  If any of you are interested in maintaining it, please let us know by commenting on this issue.  Once one of you has volunteered and there is agreement among contributors about who will take on this role, we can then reach out to rubygems and github to let them know that there is a new maintainer.

**Comment by betesh** [comment]
Since nobody else has spoken up, I will be reaching out to github and rubygems

**Comment by Empact** [comment]
@betesh FYI I use webmock for current projects.
https://github.com/bblimke/webmock

**Comment by chrisk** [comment]
Hi all, thank you for posting. I'm really sorry for leaving you hanging for so long.

When I last worked on this project, I was very sick. It eventually got serious enough that I had to stop working, pack up my apartment in San Francisco, and move home, putting my life on hold completely.

Thankfully, I eventually made it through, and I'm doing much, much better these days.

Back when I was first starting to get better, I decided to fill the newly available hours of the day with activities I thought would help shore up my physical and mental health. I started exercising again, joined a choir, started getting fluent in Spanish again, got a skateboard, started volunteering, and on and on. I spent a lot of time caring for my mother after she got sick herself and ended up in the hospital. These days, of all things, I actually get paid to sing in a professional vocal group sometimes. It's been a strange & winding path, but one I needed to walk.

All through this, I wasn't paying attention to this project, nor responding to a lot of friends who were concerned about me. I think I was pretty burned out in addition to being sick, and just wasn't ready to deal with the responsibility of working on this again, or of maintaining those relationships. So I put it all off. In hindsight, I think that was probably a healthy defense mechanism for me, but I wish I had let all of you know what was happening. Not least of all @blaine, who started this project and graciously let me start maintaining it way back in 2008. I apologize for that. I know lots of people use FakeWeb to make important things, and it's not unreasonable to expect a minimum level of communication to reciprocate your investment in a tool.

With all that out of the way, I'm here to say that I'm going to take up maintaining FakeWeb once again. You can expect a point release to restore compatibility with all current mainstream versions of Ruby. After that, I'll start on the rest of the open pull requests.

I just got a new Mac, so I'm currently setting up my environment. After I finish that you should start seeing activity on this project again.

If you need something in the meantime, certainly try WebMock and see if it works for you. That project actually started as a copy-and-paste of FakeWeb, oddly enough, although it has developed considerably since then. And if you've already switched and it works for you, then great.

Going forward, I think it's important for FakeWeb to continue to exist so that we always have a small, simple, fast tool for the 99% case of just testing code that uses Net::HTTP. Also, the gem has been an inspiration (and a template) for library authors in many other languages over the years, and that's been good for software development everywhere. I'd like to keep that going.

Okay. Whew. Thanks for reading all that, and thank you all again for your concern. If you have a fix to contribute, please do continue to open/update issues/pull requests and I'll merge. Updates should post here as I reference this ticket in commits. Onwards!

--Chris

**Comment by chrisk** [comment]
Hello everyone!

I've got a ton of fixes merged now, and as far as I know we're code-complete to release 1.3.1. I still need to update the docs and write up a changelog. In the interim, if there's anybody out there who could test FakeWeb in their projects, I'd appreciate it. The master branch should be compatible with all versions of MRI from the past decade or so (Ruby 1.8.6 or later, up to 2.4.x), as well as recent versions of JRuby and Rubinius.

You can try the current master branch by doing something like this in your Gemfile:

```ruby
group :test do
  gem "fakeweb", github: "chrisk/fakeweb", branch: "master"
end
```

**Comment by sdsykes** [comment]
I can confirm tests run ok on ruby 2.4.1 in https://github.com/sdsykes/fastimage with the latest changes. Thanks!

**Comment by choonkeat** [comment]
Perhaps a release candidate gem? Not all environments are hospitable to installing gems by git

**Comment by thewoolleyman** [comment]
@chrisk Please publish a [prerelease gem](http://guides.rubygems.org/patterns/#prerelease-gems) to rubygems.  Otherwise, your latest updates [cannot be used as a gemspec development dependency](https://stackoverflow.com/a/6499472/25192) for other gems.

**Comment by esbanarango** [comment]
@chrisk Thank you! Could we release `1.3.1` or at least have a prerelease gem? We can't use it in other gems like:

`add_development_dependency "fakeweb", github: "chrisk/fakeweb", branch: "master"`

🙏 🙏 🙏 🙏

**Comment by abarrak** [comment]
It'll be very helpful for the gem users to have `1.3.1` released.

**Comment by esbanarango** [comment]
@chrisk let me know if you need a hand here ✋ 🙏

**Comment by esbanarango** [comment]
`1.3.1` coming soon...

![xt9ignxyv1zooxxbti](https://user-images.githubusercontent.com/1156865/38534427-caf995f0-3c43-11e8-9c09-11b331f0fe36.gif)

**Comment by sdsykes** [comment]
I pushed a gem named fakeweb-fi to rubygems, built from master (version 1.3.1) - feel free to use it if you need. It has no other changes than the name of the gem and the version number - so you would still need require 'fakeweb' to load it.

It was enough to get https://github.com/sdsykes/fastimage/ passing anyway.

**Comment by esbanarango** [comment]
Thank you @sdsykes
